Client/clientCommunication.py

```diff
@@ -19,27 +19,28 @@
 from base_logger import logging
 logger = logging.getLogger(__name__)
 
+
 class ClientCommunication(GuiWriter):
     world_id: int
     game_room: str
     event_scanning: bool
     disable_multiplayer: bool
     game_handler: ClientGameConnection
 
-    def __init__(self, config: Config, signal: Signal = None):
+    def __init__(self, signal: Signal = None):
         super().__init__(signal)
+        config = Config.get_config()
```

Comment:
Please revert this change (And others like it) I prefer to pass in the `Config` through the Constructor even though it's a Singleton. 

By leaving control of the movement of the Singleton closer to the top level of the program, we make it easier to track the configuration passing through the application. Along with keeping the Logical requirements for the class to work within the Constructor. It's also easier to test with a mock config, since we can pass in anything with the same functions, instead of having to create/manipulate the actual config.

I understand that it's a matter of my personal preference, which I hate to enforce on others, but it's easier for me to maintain the code then.